packages/contracts/contracts/L1/sequencer/Sequencer.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.9.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "./ISequencer.sol";
+
+contract Sequencer is ISequencer, ReentrancyGuardUpgradeable, OwnableUpgradeable {
+    using SafeERC20 for IERC20;
+
+    mapping(address => Sequencer) public seqsDeposit;
+    address[] public depositKeys;
+    address bitToken;
+
+    function initialize(address _bitToken) public initializer {
+        __Ownable_init();
+        __ReentrancyGuard_init();
+
+        bitToken = _bitToken;
+    }
+
+    function changeBitAddress(address _bitToken) public onlyOwner {
+        bitToken = _bitToken;
+    }
+
+    function createSequencer(
+        uint256 _amount,
+        address _mintAddress,
+        bytes calldata _nodeID
+    ) external nonReentrant {
+        // check params todo: check nodeID
+        require(_amount > 0, "invild amount");
+        require(_mintAddress != address(0), "invild address, address can not be 0");
+
+        // check already have sequencer
+        require(seqsDeposit[msg.sender].mintAddress == address(0), "already have deposit");
+        IERC20(bitToken).safeTransferFrom(msg.sender, address(this), _amount);
+
+        uint256 index = depositKeys.length;
+        seqsDeposit[msg.sender] = Sequencer({
+            owner: msg.sender,
+            mintAddress: _mintAddress,
+            nodeID: _nodeID,
+            amount: _amount,
+            keyIndex: index
+        });
+        depositKeys[index] = msg.sender;
```

Comment:
Should we add an Event to notify the L2 that a new sequencer has been created?

Reply:
That`s right